kubernetes/kubernetes_service.go

```diff
@@ -238,6 +241,45 @@ func (in *IstioClient) GetPods(namespace, labelSelector string) ([]v1.Pod, error
 	}
 }
 
+// GetPod returns the pod definitions for a given pod name.
+// It returns an error on any problem.
+func (in *IstioClient) GetPod(namespace, name string) (*v1.Pod, error) {
+	if in.k8sCache != nil {
+		pods, err := in.k8sCache.GetPods(namespace)
+		if err != nil {
+			return &v1.Pod{}, err
+		}
+		for _, pod := range pods {
+			if name == pod.Name {
+				return &pod, nil
+			}
+		}
+		return &v1.Pod{}, nil
```

Comment:
same

Reply:
I'm not sure, I'll look at the code but probably you are right.

Reply:
Done